tests/integration/utils/test_sparams.py

```
@@ -0,0 +1,614 @@
+from fdtdx.objects.boundaries.perfectly_matched_layer import PerfectlyMatchedLayer
+from fdtdx.utils.sparams import (
+    PortSpec,
+    _make_port_shape,
+    calculate_sparam,
+    calculate_sparams,
+    determine_input_norm_detector_name,
+    setup_sparams_simulation,
+)
+
+# ---------------------------------------------------------------------------
+# Shared domain constants – tiny grid for speed
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)